### PR TITLE
fix: set the context when overwriting .clock command

### PR DESCRIPTION
**Pull request by Hugo-Hache**
### User facing changelog
Previously overwriting `.clock` command crashed because it did not preserve this context. Now the following works:

```js
Cypress.Commands.overwrite('clock', (originalCommand, ...args) => {
  return originalCommand(...args)
})

cy.clock()
```

### Additional details
This MR is vastly inspired from #9589

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? [Link](https://github.com/cypress-io/cypress-documentation/pull/4244)
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by Hugo-Hache** [comment]
Thanks @bahmutov for #9589, without it I would never have been able to know how to fix the crash I was facing.

File: packages/driver/cypress/integration/commands/clock_spec.js [hunk 1]
```diff
@@ -125,6 +125,14 @@ describe('src/cy/commands/clock', () => {
       })
     })
 
+    it('overwrite gracefully', () => {
```
Comment (ryanthemanuel) [inline packages/driver/cypress/integration/commands/clock_spec.js:128]: Let's be more explicit about what we're testing here. Something like:

```suggestion
    it('overwrites without crashing', () => {
```

**Comment by jennifer-shehane** [comment]
@Hugo-Hache Thanks for signing the CLA, unfortunately one of the author of the commits does not match the email address on your GitHub account. Our CLA bot gets confused when they don't match. Here are some solutions to fixing when commits are linked to the wrong user: https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
Update CLA signature

**Comment by Hugo-Hache** [comment]
@jennifer-shehane fixed the identity of my first commit to match my GitHub account

**Review by flotwig — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**